api/db-cdr/generate-cdr/make-bq-data-dictionary.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# This generates the big query data dictionary tables by importing data from
+# csv at gs://all-of-us-workbench-private-cloudsql/$BQ_DATASET/data_dictionary.
+
+set -ex
+
+export BQ_PROJECT=$1  # project
+export BQ_DATASET=$2  # dataset
+export DRY_RUN=$3     # dry run
+
+archiveList=$(gsutil ls gs://all-of-us-workbench-private-cloudsql/$BQ_DATASET/data_dictionary/archive/*.csv) > /dev/null || true;
+dataDictionaryList=$(gsutil ls gs://all-of-us-workbench-private-cloudsql/$BQ_DATASET/data_dictionary/*.csv) > /dev/null || true;
+
+if [ "$DRY_RUN" == true ]
+then
+  echo "Check if data dictionary file exists"
+  if [ -z "$dataDictionaryList" ] && [ -z "$archiveList" ]
+  then
+    echo "Error: Data Dictionary file is missing!"
+    exit 1;
+  fi
+  exit 0
+fi
+
+# Throw error if there is no data_dictionary file for CDR release
+echo "Check if data dictionary file exists"
+if [ -z "$dataDictionaryList" ] && [ -z "$archiveList" ]
```

Comment:
why do we check the archiveList as well for this error? Should we throw an error if there is more than 1 csv in the bucket? Seems like an issue if we have 2 or more at the same time?

Reply:
So i am checking archive list in case the CDR process is running the second time for the same project/dataSet. Logic returns no error if there already exist a file in archive which means the data dictionary for this has already been processed no need to keep the csv file again in the bucket.

Reply:
I can add additional error conditions to make sure dataDictionaryList has just one CSV at a time

Reply:
Done